docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+In this example I'll just add it to the class.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+
+}
+```
+
+### &lt;returns&gt;
+
+As the name suggests the `<returns>` tag is used in the comment for a method declaration to describe its return value.
+Like before this will be illustrated on the first `Add` method go ahead an implement it on other methods.
+
+```csharp
+// Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+```
+
+### &lt;value&gt;
+
+The `<value>` works similarly to the `<returns>` tag except that it is used for properties.
+Let's assume our `Math` library had a static property called `PI` here's how we'll use this tag:
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+    /// <value>Gets the value of PI.</value>
+    public static double PI { get; }
+}
+```
+
+### &lt;example&gt;
+
+The `<example>` tag lets you specify an example of how to use a class, method or other member. 
+This involves using the child `<code>` tag.
+
+```csharp
+// Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    /// <example>
+    /// <code>
+    /// int c = Math.Add(4, 5);
+    /// if (c > 10)
+    /// {
+    ///     Console.WriteLine(c);
+    /// }
+    /// </code>
+    /// </example>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+```
+
+The `code` tag preserves line breaks and indentation for longer examples.
+
+### &lt;para&gt;
+
+Sometimes there is need to format the content of certain tags and that's where the `<para>` tag comes in.
```

Comment:
passive voice. Try "You may find you need to format content in your tags. You use the `<para>` tag to format your documentation in paragraphs."